src/inmanta/data/__init__.py

```diff
@@ -390,6 +396,23 @@ def as_filter(
         :param start: is this the start filter? if so, retain all values`  > (column_value, id_value)`
 
         :return: The filter (as a string) and all associated query parameter values
+
+        None values can have a double meaning here:
+        - no value provided
+        - the value is provided and None
+
+        The distinction can be made as follows:
+        1. at least one of the columns must be not nullable (otherwise the sorting is not unique)
+        2. when both value are None, we are not paging and return '[],[]'
+        3. when one of the values is effective, we produce a filter
+
+        More specifically:
+        1. when we have a single order, and `column_value` is not None, this singe value is used for filtering
+        2. when we have a double order and the 'id_value' is not None and `is_id_nullable()`,
+            we consider the null an effective value and filter on both `column_value` and `id_value`
+        3. when we have a double order and the 'id_value' is not None and `not is_id_nullable()`,
+            we consider the null not a value and filter only on `id_value`
```

Comment:
You deleted `is_id_nullable` in this commit. Do you mean `self.get_order_by_column_type().nullable`?

Reply:
changed